resilience4j-circuitbreaker/src/main/java/io/github/resilience4j/circuitbreaker/CircuitBreaker.java

```diff
@@ -502,27 +508,27 @@ static CircuitBreaker of(String name,
      * Acquires a permission to execute a call, only if one is available at the time of invocation.
      * If a call is not permitted, the number of not permitted calls is increased.
      * <p>
-     * Returns false when the state is OPEN or FORCED_OPEN. Returns true when the state is CLOSED or
-     * DISABLED. Returns true when the state is HALF_OPEN and further test calls are allowed.
-     * Returns false when the state is HALF_OPEN and the number of test calls has been reached. If
+     * Returns Optional.empty() when the state is OPEN or FORCED_OPEN. Returns Token when the state is CLOSED or
+     * DISABLED. Returns Token when the state is HALF_OPEN and further test calls are allowed.
+     * Returns Optional.empty() when the state is HALF_OPEN and the number of test calls has been reached. If
      * the state is HALF_OPEN, the number of allowed test calls is decreased. Important: Make sure
      * to call onSuccess or onError after the call is finished. If the call is cancelled before it
      * is invoked, you have to release the permission again.
      *
-     * @return {@code true} if a permission was acquired and {@code false} otherwise
+     * @return {@code Token} if a permission was acquired and {@code Optional.empty()} otherwise
      */
-    boolean tryAcquirePermission();
+    Optional<Token> tryAcquirePermission();
```

Comment:
Even if I like `Optional` we should try to avoid it in order to reduce GC pressure.